client/lib/login/index.js

```diff
@@ -112,3 +113,44 @@ export function getSignupUrl(
 
 	return signupUrl;
 }
+
+export const getLoginLinkPageUrl = ( locale, currentRoute, isGutenboarding ) => {
+	const loginParameters = {
+		locale,
+		twoFactorAuthType: 'link',
+	};
+
+	if ( currentRoute === '/login-in/jetpack' ) {
+		loginParameters.twoFactorAuthType = 'jetpack/link';
+	} else if ( isGutenboarding ) {
+		loginParameters.twoFactorAuthType = 'new/link';
+	}
+
+	return login( loginParameters );
+};
```

Comment:
Good to see this piece of logic extracted to lib. Shouldn't we re-use it also in `LoginLinks` component? https://github.com/Automattic/wp-calypso/blob/update/login-form%2Fwrong-pwd-validation-msg/client/login/wp-login/login-links.jsx#L104-L121

Reply:
Great suggestion! I've added it in 53138a563f81928820b1d81931a4e81e2d65a6f0